pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/InternalInterfaces.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.ast;
+
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Those are some interfaces that are not published, but are used to keep
+ * uniform names on related concepts. Maybe it makes sense to publish some of
+ * them at some point.
+ */
+@SuppressWarnings("PMD.MissingStaticMethodInNonInstantiatableClass")
+final class InternalInterfaces {
+
+    private InternalInterfaces() {
+        // utility class
+    }
+
+    interface OperatorLike {
```

Comment:
Shouldn't we move these interfaces into a internal package (e.g. `net.sourceforge.pmd.lang.java.ast.internal`)?
Then they would need to be public, but we defined, that everything in "internal" is private API anyways...

Reply:
I think for this API at least, I'd rather like them to be package private. The AST API is very deep and wide, and making those supertypes public (even if internal) would make the API harder to use correctly and to learn (they would show up in autocomplete, in javadocs, etc).

Reply:
Ok, then let's keep this InternalInterfaces